protocol/communities/manager.go

```diff
@@ -1585,58 +1587,124 @@ func (m *Manager) HandleCommunityRequestToJoin(signer *ecdsa.PublicKey, request
 }
 
 type CheckPermissionToJoinResponse struct {
-	Satisfied   bool                                    `json:"satisfied"`
-	Permissions map[string]*CheckPermissionToJoinResult `json:"permissions"`
+	Satisfied         bool                                    `json:"satisfied"`
+	Permissions       map[string]*CheckPermissionToJoinResult `json:"permissions"`
+	ValidCombinations []*AccountChainIDsCombination           `json:"validCombinations"`
 }
 
 type CheckPermissionToJoinResult struct {
 	Criteria []bool `json:"criteria"`
 }
 
+type AccountChainIDsCombination struct {
+	Address  gethcommon.Address `json:"address"`
+	ChainIDs []uint64           `json:"chainIds"`
+}
+
 func (c *CheckPermissionToJoinResponse) calculateSatisfied() {
 	if len(c.Permissions) == 0 {
 		c.Satisfied = true
 		return
 	}
 
+	c.Satisfied = false
 	for _, p := range c.Permissions {
-		c.Satisfied = true
+		satisfied := true
 		for _, criteria := range p.Criteria {
 			if !criteria {
-				c.Satisfied = false
+				satisfied = false
 				break
 			}
 		}
+		if satisfied {
+			c.Satisfied = true
+		}
 	}
```

Comment:
The above here turned out to be a logical bug.
As per spec, only one permission needs to be satisfied for someone to be able to join.
This fixes the logic such that this holds true.

Prior to this change, `Satisfied` would be set to `false` if just one token criteria from any permission wasn't.